sync upstream KF into midstream ODH

tarilabs opened the pull request:
do not bot-approve :) let's ensure we can merge-commit manually :)

lampajr commented on internal/server/grpc_server_test.go:
```diff
@@ -0,0 +1,99 @@
+
+	response, err := grpcServer.PutArtifactType(context.Background(), &req)
+	if err != nil {
+		t.Logf("Should error on PutArtifactType with Request missing Name: %v", err)
```
```suggestion
		t.Logf("Expected error on PutArtifactType with Request missing Name: %v", err)
```

lampajr reviewed (approved):
lgtm

isinyaaa commented on internal/server/grpc_server_test.go:
```diff
@@ -0,0 +1,99 @@
+	defer os.Remove(f.Name())
+	db, err := setup(f)
+	if err != nil {
+		t.Errorf("Should expect DB connection: %v", err)
```
`Should expect` in error messages sounds confusing. How about:

```suggestion
		t.Errorf("Expected DB connection: %v", err)
```

isinyaaa commented on internal/server/grpc_server_test.go:
```diff
@@ -0,0 +1,99 @@
+	}
+	t.Logf("Should PutArtifactType: %v", response)
+
+	if *response.TypeId > 0 {
+		t.Logf("Should PutArtifactTypeResponse.TypeId an ID > 0: %v", response)
+	} else {
+		t.Errorf("Should PutArtifactTypeResponse.TypeId an ID > 0: %v", response)
+	}
```
If both messages are the same, why don't we
```suggestion
	t.Logf("Should PutArtifactTypeResponse.TypeId an ID > 0: %v", response)
	if *response.TypeId <= 0 {
		t.Fail()
	}
```

tarilabs replied:
I just adopted "test styleguide" from a course I've followed which tends to reuse the verbatim but use Logf Vs Errorf 🤷

isinyaaa replied:
I tend to be against this because DRY on string makes a lot of sense -- one less place to update in case the test changes, and it's not totally obvious that both strings are the exact same.

tarilabs replied:
I will adopt this other "styleguide" as I also tend to agree with DRY, until I get more clear the majority preference for the group

isinyaaa commented on internal/server/grpc_server_test.go:
```diff
@@ -0,0 +1,99 @@
+	if err != nil {
+		t.Logf("Should error on PutArtifactType with Request missing Name: %v", err)
+	} else {
+		t.Errorf("Should error on PutArtifactType with Request missing Name: %v, %v", response, err)
```
Here, `err` should be `nil`, so why print it?

isinyaaa reviewed (commented):
lgtm aside from error messages. I don't get why we're using "should ..." here every time. Some of them don't make any sense to me.

tarilabs commented:
> I don't get why we're using "should ..." here every time. Some of them don't make any sense to me.

👉  https://github.com/opendatahub-io/model-registry/pull/18#discussion_r1334532493

> Also: please use the imperative on the commit message.

I have used title within 50 chars, newline, description; I am afraid I did not understand what you meant here. Do you have an article to point me to, please.

tarilabs commented:
> Also: please use the imperative on the commit message.

👉  21743b5

dhirajsb reviewed (changes requested):
Re-use this as a unit test for the DB service handler. That way it's a quick and easy unit test and we leave integration testing to python scripts.

isinyaaa commented:
The last commit could've been a fixup. If you do squash this, please remove that message.

tarilabs commented:
> Re-use this as a unit test for the DB service handler. That way it's a quick and easy unit test and we leave integration testing to python scripts.

I think we discussed about having small unit test for "surface contacts" or "core capabilities" and serving as smoke testing was helpful.
This could have served as a small smoke/unit test in that sense.
It seems however that there is no way to have a change-of-mind about this.
Noted. ✍️ 

Rejected with https://github.com/opendatahub-io/model-registry/pull/18#pullrequestreview-1644747578 , closing PR.

lampajr committed:
kubeflow: fix go module and odh. debranding (#15)

* kubeflow: change go module name and references

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>

* kubeflow: rename odh. into kfmr.

Signed-off-by: tarilabs <matteo.mortari@gmail.com>

* kubeflow: py: pyproject description

Signed-off-by: tarilabs <matteo.mortari@gmail.com>

* kubeflow: nit picks in text documents

Signed-off-by: tarilabs <matteo.mortari@gmail.com>

---------

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>
Signed-off-by: tarilabs <matteo.mortari@gmail.com>
Co-authored-by: tarilabs <matteo.mortari@gmail.com>

lampajr committed:
Fix metadata OpenAPI spec (#17)

* fix: OpenAPI metadata discriminator

Signed-off-by: tarilabs <matteo.mortari@gmail.com>

* wiring factories and default values missed in codegen

Signed-off-by: tarilabs <matteo.mortari@gmail.com>

* introduce openapi defaults

Signed-off-by: tarilabs <matteo.mortari@gmail.com>

* fix TestMetadataValue*

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>

* fix: type assert generation

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>

* upgrade openapi spec version to v1alpha2

Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>

---------

Signed-off-by: tarilabs <matteo.mortari@gmail.com>
Signed-off-by: Andrea Lamparelli <a.lamparelli95@gmail.com>
Co-authored-by: tarilabs <matteo.mortari@gmail.com>

tarilabs committed:
kubeflow: make MLMD type names (and prefix) pluggable (#19)

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.61.1 to 1.62.0 (#20)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.61.1 to 1.62.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.61.1...v1.62.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

antisaling committed:
OAS: Fix discriminator field definition for Artifact (#22)

* gitignore: ignore all coverage files

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>

* OAS: fix discriminator field for Artifact

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>

---------

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>

tarilabs committed:
Revert "sync upstream KF into midstream ODH (opendatahub-io#15)"

This reverts commit 6ae59f5a5de531a8fd78b34997222d4480166ad5.

tarilabs committed:
Merge sync remote-tracking branch 'upstream/main' into tarilabs-20240304-syncUpstream2

dhirajsb reviewed (approved):
lgtm

tarilabs commented:
all green, proceeding with manual merge-commit (no bot)
